pkg/operators/machine-api-operator.go

```diff
@@ -85,17 +90,51 @@ var _ = Describe("[Feature:Operators] Machine API operator deployment should", f
 		client, err := framework.LoadClient()
 		Expect(err).NotTo(HaveOccurred())
 
-		Expect(framework.DeleteValidatingWebhookConfiguration(client, framework.DefaultValidatingWebhookConfiguration)).To(Succeed())
+		// Record the UID of the current ValidatingWebhookConfiguration
+		initial, err := framework.GetValidatingWebhookConfiguration(client, framework.DefaultValidatingWebhookConfiguration.Name)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(initial).ToNot(BeNil())
+		initialUID := initial.GetUID()
+
+		Expect(framework.DeleteValidatingWebhookConfiguration(client, initial)).To(Succeed())
 
+		// Ensure that either UID changes (to show a new object) or that the existing object is gone
```

Comment:
i just want to understand this a little better, shouldn't this always check that the old object was deleted?

Reply:
If the old object was deleted, and the new one created in a very quick succession, we wouldn't get a 404, so I've written this so that it can detect that the object has changed to minimise flakes if at all possible